rule/servlet/src/main/java/io/opentracing/contrib/specialagent/rule/servlet/FilterAgentRule.java

```diff
@@ -38,31 +49,32 @@
         @Override
         public Builder<?> transform(final Builder<?> builder, final TypeDescription typeDescription, final ClassLoader classLoader, final JavaModule module) {
           return builder
-            .visit(Advice.to(ServletInitAdvice.class).on(named("init").and(takesArguments(1)).and(takesArgument(0, named("javax.servlet.ServletConfig")))))
-            .visit(Advice.to(ServletServiceAdvice.class).on(named("service").and(takesArguments(2)).and(takesArgument(0, named("javax.servlet.http.HttpServletRequest")).and(takesArgument(1, named("javax.servlet.http.HttpServletResponse"))))));
+            .visit(Advice.to(ServletInitAdvice.class).on(named("init").and(takesArguments(ServletConfig.class))))
+            .visit(Advice.to(ServletServiceAdvice.class).on(named("service").and(takesArguments(HttpServletRequest.class, HttpServletResponse.class))));
         }})
       .type(not(isInterface()).and(hasSuperType(named("javax.servlet.http.HttpServletResponse"))))
       .transform(new Transformer() {
         @Override
         public Builder<?> transform(final Builder<?> builder, final TypeDescription typeDescription, final ClassLoader classLoader, final JavaModule module) {
           return builder
-            .visit(Advice.to(HttpServletResponseAdvice.class).on(named("setStatus")))
-            .visit(Advice.to(HttpServletResponseAdvice.class).on(named("sendError")));
+            .visit(Advice.to(HttpServletResponseAdvice.class).on(named("setStatus").and(takesArguments(Integer.TYPE))))
+            .visit(Advice.to(HttpServletResponseAdvice.class).on(named("sendError").and(takesArguments(Integer.TYPE, String.class))))
+            .visit(Advice.to(HttpServletResponseAdvice.class).on(named("sendError").and(takesArguments(Integer.TYPE))));
         }})
       .type(not(isInterface()).and(hasSuperType(named("javax.servlet.Filter")).and(not(named("io.opentracing.contrib.web.servlet.filter.TracingFilter")))))
       .transform(new Transformer() {
         @Override
         public Builder<?> transform(final Builder<?> builder, final TypeDescription typeDescription, final ClassLoader classLoader, final JavaModule module) {
           return builder
-            .visit(Advice.to(FilterInitAdvice.class).on(named("init").and(takesArguments(1)).and(takesArgument(0, named("javax.servlet.FilterConfig")))))
-            .visit(Advice.to(DoFilterEnter.class).on(named("doFilter")));
+            .visit(Advice.to(FilterInitAdvice.class).on(named("init").and(takesArguments(FilterConfig.class))))
+            .visit(Advice.to(DoFilterEnter.class).on(named("doFilter").and(takesArguments(ServletRequest.class, ServletResponse.class, FilterChain.class))));
```

Comment:
Same as with the first comment above, please use the `named("...")` pattern to reference the classes by name, instead of via their `.class` references.